test/src/Rect.cpp

```diff
@@ -0,0 +1,116 @@
+#include <SFML/Graphics/Rect.hpp>
+#include <SFML/System/Vector2.hpp>
+#include <catch.hpp>
+
+// Use sf::Rect<int> (IntRect) for tests. Test coverage is given, as there are no template specializations.
+
+TEST_CASE("sf::Rect class")
+{
+    SECTION("Construction")
+    {
+        SECTION("Default constructor")
+        {
+            sf::IntRect rectangle;
+            CHECK(rectangle.left == 0);
+            CHECK(rectangle.top == 0);
+            CHECK(rectangle.width == 0);
+            CHECK(rectangle.height == 0);
+        }
+
+        SECTION("(left, top, width, height) constructor")
+        {
+            sf::IntRect rectangle(1, 2, 3, 4);
+            CHECK(rectangle.left == 1);
+            CHECK(rectangle.top == 2);
+            CHECK(rectangle.width == 3);
+            CHECK(rectangle.height == 4);
+        }
+
+        SECTION("(Vector2, Vector2) constructor")
+        {
+            sf::Vector2i position(1, 2);
+            sf::Vector2i dimension(3, 4);
+            sf::IntRect rectangle(position, dimension);
+
+            CHECK(rectangle.left == 1);
+            CHECK(rectangle.top == 2);
+            CHECK(rectangle.width == 3);
+            CHECK(rectangle.height == 4);
+        }
+
+        SECTION("Conversion constructor")
+        {
+            sf::FloatRect sourcerectangle(1.0f, 2.0f, 3.0f, 4.0f);
+            sf::IntRect rectangle(sourcerectangle);
+
+            CHECK(rectangle.left == static_cast<int>(sourcerectangle.left));
+            CHECK(rectangle.top == static_cast<int>(sourcerectangle.top));
+            CHECK(rectangle.width == static_cast<int>(sourcerectangle.width));
+            CHECK(rectangle.height == static_cast<int>(sourcerectangle.height));
+        }
+    }
+
+    SECTION("Containment")
+    {
+        SECTION("contains(x, y)")
+        {
+            sf::IntRect rectangle(0, 0, 10, 10);
+
+            CHECK(rectangle.contains(0, 0) == true);
+            CHECK(rectangle.contains(10, 10) == false);
+            CHECK(rectangle.contains(15, 15) == false);
```

Comment:
Maybe more cases at the corner/edge, e.g.: (0, 9), (0, 10), (9, 9), (9, 10) -- and symmetric with width/height?